cognite/neat/_utils/cdf/loaders/_data_modeling.py

```diff
@@ -206,45 +209,70 @@ def are_equal(self, local: ViewApply, remote: View) -> bool:
     def as_write(self, view: View) -> ViewApply:
         return ViewApply.load(self._as_write_raw(view))
 
-    def retrieve_all_ancestors(self, views: list[ViewId]) -> ViewList:
-        return ViewList(self._retrieve_view_ancestors(views, self._cache_view_by_id))
+    def retrieve_all_ancestors(self, views: list[ViewId], include_connections: bool = False) -> ViewList:
+        return ViewList(self._retrieve_view_ancestors(views, include_connections, self._cache_view_by_id))
 
-    def _retrieve_view_ancestors(self, parents: list[ViewId], cache: dict[ViewId, View]) -> list[View]:
+    def _retrieve_view_ancestors(
+        self, parents: list[ViewId], include_connections: bool, cache: dict[ViewId, View]
+    ) -> list[View]:
         """Retrieves all ancestors of a view.
 
         This will mutate the cache passed in, and return a list of views that are the ancestors
         of the views in the parents list.
 
         Args:
             parents: The parents of the view to retrieve all ancestors for
+            include_connections: Whether to include all sources.
             cache: The cache to store the views in
         """
-        parent_ids = parents.copy()
+        next_backlog_ids = parents.copy()
         found: list[View] = []
         found_ids: set[ViewId] = set()
-        while parent_ids:
+        while next_backlog_ids:
```

Comment:
does DMS have limit on inheritance depth?

Reply:
not as far as I know